### PR TITLE
🔖 Release workflow

### PR DESCRIPTION
**Pull request by je-cook**
## Linked Issues

See #1087 

## Description

Create a workflow document that can be followed to create a new release

TODO

- [ ] Where should this live? in the base of the repo as an md file? in the docs as an rst?
- [ ] Look into creating a github action to make a release
- [ ] Look into using something like towncrier (https://github.com/twisted/towncrier) to create an automatic CHANGELOG
- [ ] Possibility of auto creating a tag...?

## Checklist

I confirm that I have completed the following checks:

- [ ] Tests run locally and pass `pytest tests --reactor`
- [ ] Code quality checks run locally and pass `flake8` and `black .`
- [ ] Documentation built locally and checked `sphinx-build -W documentation/source documentation/build`

### CODE REVIEW

**Comment by je-cook** [comment]
Some thoughts I had:

Is it possible to force a straight fast forward merge as opposed to squash and merge just in case some accidentally forgets to press the down arrow?

I think you've got a trailing '\n' in your PR body, nice try on the logs thing pity it didnt work

File: scripts/version_checker.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+"""
+Validate an input version string using the packaging.version module, and
```
Comment (je-cook) [inline scripts/version_checker.py:2]: This file can probably be removed now

File: .github/workflows/check_release_due.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+            echo "Release is not due"
+          fi
+      - name: Checkout
+        uses: actions/checkout@v3
+        if: ${{ steps.check-date.outputs.release_due == 'true' }}
+      - name: Create Issue
+        id: create-issue
+        if: ${{ steps.check-date.outputs.release_due == 'true' }}
+        run: >
+          gh issue create \
```
Comment (je-cook) [inline .github/workflows/check_release_due.yml:34]: Just a note for the conversation we're having elsewhere:

We could/should ff merge for the develop_dependencies too also separately to this we should find a way torebase develop_dependencies and resign the relevant commits

**Review by je-cook — COMMENTED** [review]
Looks good! 

Just the develop_dependencies conversation

**Comment by je-cook** [comment]
Well done! I think we need to add the status check to the required to pass on the develop branch protection rules as well.

**Comment by hsaunders1904** [comment]
After discussion offline, we are changing tac here and are abandoning the idea of a fast-forward merge workflow. The permissions were getting too fiddly to implement, and the release workflow was getting very complicated for what should be a simple process.

Closing this in favour of #1126.

**Pull request closed without merging**